dev-client/src/model/soilId/actions/soilDataDiff.test.ts

```diff
@@ -15,71 +15,388 @@
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: '', depthInterval: {start: 2, end: 3}},
+      ];
+      prev.depthIntervals = [
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: '', depthInterval: {start: 2, end: 3}},
+      ];
+
+      const deleted = getDeletedDepthIntervals(curr, undefined);
+      expect(deleted).toEqual([]);
+    });
+
+    test('returns prev depth intervals when deleted', () => {
+      prev.depthIntervals = [
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: '', depthInterval: {start: 2, end: 3}},
+      ];
+
+      const deleted = getDeletedDepthIntervals(curr, prev);
+      expect(deleted).toEqual([
+        {start: 1, end: 2},
+        {start: 2, end: 3},
+      ]);
+    });
+
+    test('returns only deleted prev intervals', () => {
+      curr.depthIntervals = [{label: '', depthInterval: {start: 2, end: 3}}];
+      prev.depthIntervals = [
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: '', depthInterval: {start: 2, end: 3}},
+      ];
+
+      const deleted = getDeletedDepthIntervals(curr, prev);
+      expect(deleted).toEqual([{start: 1, end: 2}]);
+    });
   });
 
-  test('returns empty when no deleted records', () => {
-    curr.depthIntervals = [
-      {label: '', depthInterval: {start: 1, end: 2}},
-      {label: '', depthInterval: {start: 2, end: 3}},
-    ];
-    prev.depthIntervals = [
-      {label: '', depthInterval: {start: 1, end: 2}},
-      {label: '', depthInterval: {start: 2, end: 3}},
-    ];
-
-    const deleted = getDeletedDepthIntervals(curr, undefined);
-    expect(deleted).toEqual([]);
+  describe('getChangedDepthIntervals', () => {
+    let curr: SoilData;
+    let prev: SoilData;
+
+    beforeEach(() => {
+      curr = {
+        depthIntervalPreset: 'CUSTOM',
+        depthIntervals: [],
+        depthDependentData: [],
+      };
+      prev = {
+        depthIntervalPreset: 'CUSTOM',
+        depthIntervals: [],
+        depthDependentData: [],
+      };
+    });
+
+    test('returns all depth intervals when no previous record', () => {
+      curr.depthIntervals = [
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: '', depthInterval: {start: 2, end: 3}},
+      ];
+
+      const changed = getChangedDepthIntervals(curr, undefined);
+      expect(changed).toHaveLength(2);
+      expect(changed[0].depthInterval).toEqual({start: 1, end: 2});
+      expect(changed[1].depthInterval).toEqual({start: 2, end: 3});
+    });
+
+    test('returns only changed intervals', () => {
+      curr.depthIntervals = [
+        {label: 'changed', depthInterval: {start: 1, end: 2}},
+        {label: 'old', depthInterval: {start: 2, end: 3}},
+        {label: 'added', depthInterval: {start: 3, end: 4}},
+      ];
+      prev.depthIntervals = [
+        {label: '', depthInterval: {start: 1, end: 2}},
+        {label: 'old', depthInterval: {start: 2, end: 3}},
+      ];
```

Comment:
Could add a 'removed' one just to confirm we won't count removed depthIntervals

Reply:
Wil do.